cpp/log/etcd_consistent_store-inl.h

```diff
@@ -625,14 +619,18 @@ void EtcdConsistentStore<Logged>::OnEtcdServingSTHUpdated(
 
 
 template <class Logged>
-util::Status EtcdConsistentStore<Logged>::RunOneCleanUpIteration(
-    const int64_t clean_up_to_sequence_number) {
-  CHECK_LE(0, clean_up_to_sequence_number);
+util::Status EtcdConsistentStore<Logged>::CleanupOldEntries() {
   if (!election_->IsMaster()) {
     return util::Status(util::error::PERMISSION_DENIED,
                         "Non-master node cannot run cleanups.");
   }
 
+  if (!serving_sth_) {
```

Comment:
Here and at line 641, you're accessing ```serving_sth_``` without holding ```mutex_```? Isn't that wrong? Could probably deal with this by calling ```GetServingSTH```, which would give you a copy (or tell you that there's no serving STH).